hogvm/typescript/src/utils.ts

```diff
@@ -79,3 +82,29 @@ export function convertHogToJS(x: any): any {
     }
     return x
 }
+
+export function calculateCost(object: any): any {
```

Comment:
Is this not going to be itself very expensive to run?

Reply:
There's always the tradeoff of safety vs performance. I think this will be quite fast. We only run it once when something goes on the stack. Objects/arrays will of course take time to loop over, but 🤷.

We will however be wrong at times, and possibly by a few times. For example If we have an object `let o := {a: {b: {c: 'str'}}}`, then when we run `o.a` in `o.a.b.c`, we add the entire `{b: {c: 'str'}}` object to the stack in addition to `o` itself that's still there. We then count its size as if it was a new object. Internally JS is for sure sharing pointers and not duplicating memory.

However given the limited scope of what we're doing here, this might just be fine.

Reply:
If you say so. I can't say for sure other than "interesting" 😅

Also does it protect against a reference object? I don't actually know how we are doing state modifications but if I were to set a nested property to be the value of the parent does is that protected against?

Reply:
Globals can't be modified, globals when read are always copied. Local variables are just elements on the stack.